Mathlib/Analysis/Analytic/Order.lean

```diff
@@ -11,91 +11,152 @@ import Mathlib.Analysis.Analytic.IsolatedZeros
+    {n : ℕ} :
+    analyticOrderNatAt f z₀ = n ↔
+      ∃ (g : 𝕜 → E), AnalyticAt 𝕜 g z₀ ∧ g z₀ ≠ 0 ∧ ∀ᶠ z in 𝓝 z₀, f z = (z - z₀) ^ n • g z := by
+  simp [← Nat.cast_inj (R := ℕ∞), Nat.cast_analyticOrderNatAt hf', hf.analyticOrderAt_eq_natCast]
 
 /-- The order of an analytic function `f` at `z₀` is finite iff `f` can locally be written as `f z =
-  (z - z₀) ^ order • g z`, where `g` is analytic and does not vanish at `z₀`.
+  (z - z₀) ^ analyticOrderNatAt f z₀ • g z`, where `g` is analytic and does not vanish at `z₀`.
 
 See `MeromorphicNFAt.order_eq_zero_iff` for an analogous statement about meromorphic functions in
 normal form.
 -/
-lemma order_ne_top_iff (hf : AnalyticAt 𝕜 f z₀) :
-    hf.order ≠ ⊤ ↔ ∃ (g : 𝕜 → E), AnalyticAt 𝕜 g z₀ ∧ g z₀ ≠ 0
-      ∧ f =ᶠ[𝓝 z₀] fun z ↦ (z - z₀) ^ (hf.order.toNat) • g z := by
-  simp only [← ENat.coe_toNat_eq_self, Eq.comm, EventuallyEq, ← hf.order_eq_nat_iff]
+lemma AnalyticAt.analyticOrderAt_ne_top (hf : AnalyticAt 𝕜 f z₀) :
+    analyticOrderAt f z₀ ≠ ⊤ ↔
+      ∃ (g : 𝕜 → E), AnalyticAt 𝕜 g z₀ ∧ g z₀ ≠ 0 ∧
+        f =ᶠ[𝓝 z₀] fun z ↦ (z - z₀) ^ analyticOrderNatAt f z₀ • g z := by
+  simp only [← ENat.coe_toNat_eq_self, Eq.comm, EventuallyEq, ← hf.analyticOrderAt_eq_natCast,
+    analyticOrderNatAt]
+
+@[deprecated (since := "2025-05-03")]
+alias AnalyticAt.order_ne_top_iff := AnalyticAt.analyticOrderAt_ne_top
 
 @[deprecated (since := "2025-02-03")]
-alias order_neq_top_iff := order_ne_top_iff
+alias AnalyticAt.order_neq_top_iff := AnalyticAt.analyticOrderAt_ne_top
+
+lemma analyticOrderAt_eq_zero : analyticOrderAt f z₀ = 0 ↔ ¬ AnalyticAt 𝕜 f z₀ ∨ f z₀ ≠ 0 := by
+  by_cases hf : AnalyticAt 𝕜 f z₀
+  · rw [← ENat.coe_zero, hf.analyticOrderAt_eq_natCast]
+    constructor
+    · intro ⟨g, _, _, hg⟩
+      simpa [hf, hg.self_of_nhds]
+    · exact fun hz ↦ ⟨f, hf, hz.resolve_left <| not_not_intro hf, by simp⟩
+  · simp [hf]
+
+lemma analyticOrderAt_ne_zero : analyticOrderAt f z₀ ≠ 0 ↔ AnalyticAt 𝕜 f z₀ ∧ f z₀ = 0 := by
+  simp [analyticOrderAt_eq_zero]
 
 /-- The order of an analytic function `f` at `z₀` is zero iff `f` does not vanish at `z₀`. -/
-lemma order_eq_zero_iff (hf : AnalyticAt 𝕜 f z₀) :
-    hf.order = 0 ↔ f z₀ ≠ 0 := by
-  rw [← ENat.coe_zero, hf.order_eq_nat_iff]
-  constructor
-  · intro ⟨g, _, _, hg⟩
-    simpa [hg.self_of_nhds]
-  · exact fun hz ↦ ⟨f, hf, hz, by simp⟩
+protected lemma AnalyticAt.analyticOrderAt_eq_zero (hf : AnalyticAt 𝕜 f z₀) :
+    analyticOrderAt f z₀ = 0 ↔ f z₀ ≠ 0 := by simp [hf, analyticOrderAt_eq_zero]
+
+@[deprecated (since := "2025-05-03")]
+alias AnalyticAt.order_eq_zero_iff := AnalyticAt.analyticOrderAt_eq_zero
+
+/-- The order of an analytic function `f` at `z₀` is zero iff `f` does not vanish at `z₀`. -/
+protected lemma AnalyticAt.analyticOrderAt_ne_zero (hf : AnalyticAt 𝕜 f z₀) :
+    analyticOrderAt f z₀ ≠ 0 ↔ f z₀ = 0 := hf.analyticOrderAt_eq_zero.not_left
 
 /-- An analytic function vanishes at a point if its order is nonzero when converted to ℕ. -/
```

Comment:
This docstring doesn't match the content very well any more (analyticity isn't required). I also suggest adding a formulation with `analyticOrderAt`.
```suggestion
/-- An function vanishes at a point if its analytic order is nonzero in `ℕ∞`. -/
lemma apply_eq_zero_of_analyticOrderAt_ne_zero (hf : analyticOrderAt f z₀ ≠ 0) :
    f z₀ = 0 := by
  by_cases hf' : AnalyticAt 𝕜 f z₀ <;> simp_all [analyticOrderAt_eq_zero] 

/-- A function vanishes at a point if its analytic order is nonzero when converted to 
ℕ. -/
```